xonsh/tools.py

```diff
@@ -1582,9 +1582,13 @@ def expanduser_abs_path(inp):
 def expand_case_matching(s):
```

Comment:
the decision is your @Cheaterman ! Manual per character parsing in python is hella slow, cause you have to create a lot of intermediate objects, regex parsing is done on the C level.